gym_graph_traffic/envs/segment.py

```diff
@@ -93,23 +93,41 @@ def update_first_phase(self) -> None:
         First phase of segment update: cellular automata step, and (sometimes) passing car to following segment.
         """
 
-        # extend p vector by free cells of following segment
-        next_segment_free_cells = self.next_intersection.can_i_go(self.idx)
-        if next_segment_free_cells > 0:
-            self.p = np.append(self.p, np.zeros(next_segment_free_cells))
-
-        # update cellular automata
-        self._nagel_schreckenberg_step()
+        info_can_i_go = None
+        # check if there is auto (1) in the last 5 cells of the segment
+        if 1 in self.p[-5:]:
+            # get car indices from the last cells
+            cars_indices = [i for i in self.p.nonzero()[0] if i >= 95]
```

Comment:
Shouldn't these -5 and 95 constants be some configuration parameters instead?

Reply:
You are probably right, as long as the parameter '-5' is for the last five cells in the p vector, the parameter '95' must be changed, because that won't work if we change the length of the segment, e.g. from 100 to 200.